### PR TITLE
Enhancing the SqlDbType with Json

### LINKED ISSUE #103925: Expose Json in SqlDbType
### Background and motivation

The `System.Data.SqlDbType` enum represents the datatypes supported by SQL Server and is used with SqlParameter to specify the column type to be used in SQL server operations while execution `SqlCommand`.

With the JSON datatype being supported in SQL Server [link](https://learn.microsoft.com/en-us/sql/t-sql/data-types/json-data-type?view=azuresqldb-current) There is a need to support the JSON type in Microsoft.Data.SqlClient ADO.Net provider for SQL Server.

The API suggestion is aimed at adding an enum called `Json` with value `35` in `SqlDbType`. 
Once this enum is available Microsoft.Data.SqlClient (the SQL Server driver) can then leverage the enum value to allow JSON operations using Microsoft.Data.SqlClient APIs.

```C#

namespace System.Data
{
    // Specifies the SQL Server data type.
    public enum SqlDbType
    {
        Json = 35,
    }
}
```

The version of Microsoft.Data.SqlClient targeting .Net 9, will be able to use the enum `SqlDbType.Json` to provide JSON support.

### API Proposal

```C#

namespace System.Data
{
    // Specifies the SQL Server data type.
    public enum SqlDbType
    {
        Json = 35,
    }
}
```

### API Usage

```csharp
using Microsoft.Data.SqlClient;


SqlParameter param = new SqlParameter();

param.SqlDbType = System.Data.SqlDbType.Json;

```

There are parameters / properties in the constructors and APIs, which will be enhanced to support `SqlDbType.Json` 

`SqlParameter`
`SqlParameterCollection`
`SqlMetaData`

### Alternative Designs

_No response_

### Risks

_No response_

### PR DESCRIPTION
**Pull request by saurabh500**
Fixes https://github.com/dotnet/runtime/issues/103925 

The API review is approved. 

The docs changes will need to be made in https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Data/SqlDbType.xml and can be separated from this PR.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
You'll need to add this member to the enum in the corresponding ref assembly `System.Data.Common/ref/System.Data.Common.cs` as well.

**Comment by saurabh500** [comment]
Thanks @jkoritzinsky  I will update the ref assembly. What about any docs related changes?

**Comment by jkoritzinsky** [comment]
For docs, I believe you can put XML comments on the change in the src/ folder and they should propagate to docs when the next version with your change ships.

@carlossanlop can confirm.

**Comment by saurabh500** [comment]
@roji this is ready for review. I have answered the question about the docs in the PR description.

File: src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs [hunk 1]
```diff
@@ -38,5 +38,6 @@ public enum SqlDbType
         Time = 32,
         DateTime2 = 33,
         DateTimeOffset = 34,
+        Json = 35,
```
Comment (stephentoub) [inline src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs:41]: Nit: I realize none of the other values have XML comments, but in general we use XML comments to seed the docs. I'd suggest adding one here. (For bonus points but not required, you could make the whole type consistent by copying the XML comments from the docs for all the other values. :)

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs (hunk 1 above)
Comment (saurabh500) [reply src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs:41]: @stephentoub I will add the XML docs for all the types here. The only trouble I have is with linking of these XML docs with the other types. The doc pages show that the comments of each of these enums link to types like `SqlXml` or `Array` etc. 
Not sure how that works. But I agree that this file could benefit from atleast adding the XML docs. I will send an update to the PR.

Comment (stephentoub) [reply src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs:41]: > I will add the XML docs for all the types here.

Ok, thanks :)

> Not sure how that works.

e.g. `<see cref="Array" />`

Comment (saurabh500) [reply src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs:41]: Great. That's not too bad then. The existing XML file in the docs repo, already has the right format, which can be brought over. 
e.g. 
```
       <summary>An XML value. Obtain the XML as a string using the <see cref="M:System.Data.SqlClient.SqlDataReader.GetValue(System.Int32)" /> method or <see cref="P:System.Data.SqlTypes.SqlXml.Value" /> property, or as an <see cref="T:System.Xml.XmlReader" /> by calling the <see cref="M:System.Data.SqlTypes.SqlXml.CreateReader" /> method.</summary>

```

**Comment by saurabh500** [comment]
cc @apoorvdeshmukh  @deepaksa1 for your awareness

File: src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs [hunk 2]
```diff
@@ -3,40 +3,185 @@
...
+
+        /// <summary>
+        /// Date and time data with time zone awareness. Date value range is from January 1,1 AD through December 31, 9999 AD.
+        /// Time value range is 00:00:00 through 23:59:59.9999999 with an accuracy of 100 nanoseconds. Time zone value range
+        /// is -14:00 through +14:00.
+        /// </summary>
         DateTimeOffset = 34,
+
+        /// <summary>
+        /// A Json Value.
```
Comment (roji) [inline src/libraries/System.Data.Common/src/System/Data/SqlDbType.cs:183]: For consistency with the other comments, for example, XML.

```suggestion
        /// A JSON value.
```

**Review by roji — APPROVED** [review]

**Pull request merged**